src/states/AdsDone.js

```diff
@@ -0,0 +1,19 @@
+import videojs from 'video.js';
+
+import {ContentState} from '../states.js';
+
+export default class AdsDone extends ContentState {
+
+  init(player) {
+    // From now on, `ended` events won't be redispatched
+    player.ads._contentHasEnded = true;
+  }
+
+  /*
+   * Midrolls do not play after ads are done.
+   */
+  startLinearAdMode() {
+    videojs.log.warn('Unexpected startLinearAdMode invocation (AdsDone)');
+  }
```

Comment:
It logically makes sense to trigger `ended` in this state to me...what do you think?

Reply:
I'm not sure it's safe to do that. For example, one of the ended events happens in a 1 millisecond timeout, but not others. I think this is a good area of future improvement, but I don't think it can be safely changed in this PR without lots of re-testing.